src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -186,6 +186,15 @@ class boss_eadric : public CreatureScript
             Initialize();
         }
 
+        void SpellHit(Unit* /*caster*/, const SpellInfo *spell) override
+        {
+            if (IsHeroic())
+                if (spell->Id == 66905)
+                    if (me->GetHealth() <= (uint32)spell->GetEffect(0)->BasePoints)
+                        if (InstanceScript* instance = me->GetInstanceScript())
+                            instance->DoUpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_BE_SPELL_TARGET, SPELL_EADRIC_ACHIEVEMENT);
```

Comment:
Multiple hacks
1) Heroic condition should be set at achievement_criteria_data
2) Use a spell to trigger the criteria (there's a spell for it - the one you're using)